src/System.Time/System/Date.cs

```diff
@@ -1721,33 +1728,40 @@ private static string NormalizeDateFormat(string format)
             // standard formats
             if (format.Length == 1)
             {
-                // pass-through formats
-                if ("DdMmYy".Contains(format))
+                switch (format)
                 {
```

Comment:
This will actually cause format.GetHashCode to be called. You should probably switch on format[0] to get a char and have the case statements be for chars.

Reply:
Fixed.  Thanks.